packages/selenium/test/RunPartial/TestBase.cs

```diff
@@ -0,0 +1,166 @@
+using System;
+using System.IO;
+using System.Reflection;
+using System.Text;
+using Deque.AxeCore.Commons;
+using Deque.AxeCore.Selenium;
+using NUnit.Framework;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using System.Threading;
+using WebDriverManager;
+using WebDriverManager.DriverConfigs;
+using WebDriverManager.DriverConfigs.Impl;
+
+namespace Deque.AxeCore.Selenium.Test.RunPartial
+{
+    [TestFixture]
+    [NonParallelizable]
+    [Category("RunPartial")]
+    public abstract class TestBase
+    {
+        private static string ChromeDriverPath = null;
+        protected IWebDriver driver { get; set; }
+
+        protected string axeSource { get; set; } = null;
+        protected string axeCrashScript { get; set; } = null;
+        protected string dylangConfigPath { get; set; } = null;
+        protected string axeRunPartialThrows { get; set; } =
+            ";axe.runPartial = () => { throw new Error(\"No runPartial\")}";
+
+        [OneTimeSetUp]
+        public void OneTimeSetup()
+        {
+            var assembly = typeof(TestBase).GetTypeInfo().Assembly;
+            axeSource = new BundledAxeScriptProvider().GetScript();
+
+            axeCrashScript = File.ReadAllText(
+                Path.Combine(TestContext.CurrentContext.TestDirectory, "fixtures/axe-crasher.js")
+            );
+
+            dylangConfigPath = FixturePath("dylang-config.json");
+
+            driver = NewDriver();
+        }
+
+        [OneTimeTearDown]
+        public void TearDown()
+        {
+            driver.Close();
+            driver.Quit();
+        }
+
+        protected AxeBuilderOptions CustomSource(string axeSource)
+        {
+            return new AxeBuilderOptions
+            {
+                ScriptProvider = new StringAxeScriptProvider(axeSource)
+            };
+        }
+        protected void GoToResource(string url)
+        {
+            driver.Navigate().GoToUrl(ResourceUrl(url));
+        }
+
+        protected void GoToFixture(string url)
+        {
+            driver.Navigate().GoToUrl(FixtureUrl(url));
+        }
+
+        protected void GoToUrl(string url)
+        {
+            driver.Navigate().GoToUrl(url);
+        }
+
+        protected void AssertHasViolations(AxeResult results, params string[] rules)
```

Comment:
You can make this much simpler (and give it better error messages on failures) by replacing this whole method with [FluentAssertions' `BeEquivalentTo`](https://fluentassertions.com/collections/):

```csharp
results.Violations.Select(v => v.Id).Should().BeEquivalentTo(rules);
```

Reply:
(that said, it doesn't look like this method is actually called anywhere, so maybe preferable to just remove it)